http/middleware.go

```diff
@@ -160,15 +160,34 @@ type AuthorizationParams struct {
 // authorization options.
 type AuthorizationOption func(*AuthorizationParams) error
 
-// AuthorizedParty sets the authorized parties that will be checked
-// against the azp JWT claim.
-func AuthorizedParty(parties ...string) AuthorizationOption {
+// AuthorizedParty allows to provide a handler that accepts the
+// 'azp' claim.
+// The handler can be used to perform validations on the azp claim
+// and should return false to indicate that something is wrong.
+func AuthorizedParty(handler func(string) bool) AuthorizationOption {
 	return func(params *AuthorizationParams) error {
-		params.SetAuthorizedParties(parties...)
+		params.AuthorizedPartyFunc = handler
 		return nil
 	}
 }
 
+// AuthorizedPartyFunc registers a handler that checks that the
+// 'azp' claim's value is included in the provided parties.
+func AuthorizedPartyFunc(parties ...string) func(string) bool {
```

Comment:
As discussed in Slack, this personally confuses me.

Maybe a naming that conveys what it does (check containment in list) instead of a name that coincides with the param below.

Reply:
@yourtallness I made a couple of changes in https://github.com/clerk/clerk-sdk-go/pull/247/commits/a23a3917a1afc471ddd169b0c6a19bce843f3eb4

- Renamed the verification parameter from AuthorizedPartyFunc to AuthorizedPartyHandler
- Renamed the option helper in the middleware from AuthorizedPartyFunc to AuthorizedPartyMatches

Please let me know what you think, thanks!

Reply:
This is an improvement.

The ideal UX would probably automatically call the default matcher if no handler is provided, but I understand this may be impractical.

So, approving AFAIAC.